sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ScalaUdf.scala

```diff
@@ -66,7 +66,7 @@ case class ScalaUdf(function: AnyRef, dataType: DataType, children: Seq[Expressi
       val child0 = children(0)
       (input: Row) => {
         func(
-          ScalaReflection.convertToScala(child0.eval(input), child0.dataType))
+          CatalystTypeConverters.convertToScala(child0.eval(input), child0.dataType))
```

Comment:
Could we use `createToScalaConverter` once as a lazy val in the body of this expression and then use the generated converter for each invocation.  This is probably a path we care about performance for.